src/protocol/event.rs

```diff
@@ -188,20 +189,28 @@ pub struct Event {
     pub received: Annotated<DateTime<Utc>>,
 
     /// Server or device name the event was generated on.
-    #[metastructure(pii_kind = "hostname")]
+    #[metastructure(pii_kind = "hostname", max_chars = "symbol")]
     pub server_name: Annotated<String>,
 
     /// Program's release identifier.
-    // TODO: cap size
-    pub release: Annotated<String>,
+    #[metastructure(max_chars = "symbol", match_regex = r"^[^\r\n]*\z")]
+    pub release: Annotated<LenientString>,
 
     /// Program's distribution identifier.
-    // TODO: cap size
-    pub dist: Annotated<String>,
+    // Match whitespace here, which will later get trimmed
+    #[metastructure(
+        max_chars = "symbol",
+        match_regex = r"^\s*[a-zA-Z0-9_.-]+\s*$"
+    )]
+    pub dist: Annotated<LenientString>,
 
     /// Environment the environment was generated in ("production" or "development").
-    // TODO: cap size
-    pub environment: Annotated<String>,
+    #[metastructure(max_chars = "symbol", match_regex = r"^[^\r\n\x0C/]+$")]
+    pub environment: Annotated<LenientString>,
```

Comment:
I would make this `"enumlike"`. 

More importantly, I think we should not permit lenient strings here and rather reject any non-string. This is certainly an error if anything else arrives here and there is no value in keeping bogus values around. Same applies to `dist` and `release`.

Reply:
There are tests in EventManager explicitly testing for this and I don't see the harm in stringifying. It's just compat breakage.